src/dinit.cc

```diff
@@ -802,13 +830,26 @@ static void control_socket_cb(eventloop_t *loop, int sockfd) noexcept
     }
 }
 
+static void control_socket_ready() noexcept {
+    if (!control_socket_open || socket_ready_fd < 0) {
+        return;
+    }
+    write(socket_ready_fd, control_socket_path, strlen(control_socket_path) + 1);
```

Comment:
We may as well close `socket_ready_fd` after writing to it here, I think

Reply:
generally yes, but that will make things go wrong if you use a standard descriptor (stdout/stderr) to print out the socket